Manifold/Error.swift

```diff
@@ -20,12 +37,24 @@ public enum Error: Printable, StringLiteralConvertible {
 	// MARK: Printable
 
 	public var description: String {
-		switch self {
-		case let Leaf(reason):
-			return reason
-		case let Branch(errors):
-			return join("\n", lazy(errors).map(toString))
-		}
+		return analysis(
+			ifLeaf: id,
+			ifBranch: { "\n".join(lazy($0).map(toString)) })
+	}
+
+
+	// MARK: StringInterpolationConvertible
+
+	public init(stringInterpolation strings: Error...) {
+		self = Error(reason: reduce(strings, "") {
+			$0 + $1.analysis(
+				ifLeaf: id,
+				ifBranch: const(""))
+		})
+	}
+
+	public init<T>(stringInterpolationSegment expr: T) {
+		self = Error(reason: toString(expr))
```

Comment:
`StringInterpolationConvertible` is required to be able to do `let e: Error = "foo: \(bar)"` as well as `let e: Error = "foo"`.